FreeRTOS_IP.c

```diff
@@ -2356,7 +2360,8 @@ static eFrameProcessingResult_t prvProcessIPPacket( IPPacket_t * pxIPPacket,
  * @return eReleaseBuffer when the message buffer should be released, or eReturnEthernetFrame
  *                        when the packet should be returned.
  */
-    static eFrameProcessingResult_t prvProcessICMPPacket( ICMPPacket_t * const pxICMPPacket )
+    static eFrameProcessingResult_t prvProcessICMPPacket( ICMPPacket_t * const pxICMPPacket,
+                                                          NetworkBufferDescriptor_t * const pxNetworkBuffer )
     {
         eFrameProcessingResult_t eReturn = eReleaseBuffer;
 
```

Comment:
actually that could be an assertion, since the data length should be checked in
`usGenerateProtocolChecksum()` or `xCheckSizeFields()` dependent on
`ipconfigDRIVER_INCLUDED_RX_IP_CHECKSUM`
but when I look at the code in ipv6 branch you mentioned,
I miss the check for the destination address to be the local ip address.
Looks like it will answer broadcast pings?
Will it use the broadcast address in the response to such pings?

Reply:
the assertion uncovered the problem that I suspected previously:
https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/237#issuecomment-832679396
> The only thing that is odd in the test harnesh is that
pxIPPacket should alias pxNetworkBuffer->pucEthernetBuffer but it isn't.

fixed the test harnesh, and it works now.
so the assertion gives the cbmc something interesting to prove :-)

Reply:
okay looks now almost like in the ipv6 branch, but I left the assertion, because I like it double-safe...